recipes-omnect/images/omnect-os-image.bb

```diff
@@ -118,3 +118,72 @@ python () {
 }
 
 inherit omnect_user
+
+inherit logging
+
+# postive test for packages, i.e. check if installed
+check_installed_packages() {
+    local manifest="$1"
+    local pkglist="$2"
+    local ret=0
+    for p in $pkglist; do
+        if ! grep -q "^$p " "$manifest"; then
+            bberror "Required TOOL not installed in image: $p"
+            ret=1
+         fi
+    done
+    return $ret
+}
+
+# negative test for packages, i.e. check if _not_ installed
+check_uninstalled_packages() {
```

Comment:
isn't that the exact function as check_installed_packages without negating the result of grep? so i could negate the result of check_installed_packages?

Reply:
nope, both functions fail already for one package not meeting the condition, which is only identical to failing for all packages if the list contains only one packet.

Reply:
:thumbsup:
what if such a packet is already a dependency of a system package?

Reply:
that is a good question, but currently I don't have another answer than that we will have to talk about the regression test again then ;-)
on the other hand I can't imagine such a dependency that we would want to overrule.
or put in other words: if such a dependency arises it doesn't make sense anymore to include such packets in OMNECT_DEVEL_TOOLS what would fail for release images. and OMNECT_TOOLS isn't critical in this respect for its nature of biegn treated as a positive list only.

Reply:
if bsp A depends on a package and bsp B doesnt ... removing the package from OMNECT_(DEVEL)_TOOLS seems not to be feasible. 
imho we should test that all package are installed we want.
for testing if packages are not installed in case of release vs OMNECT_DEVEL_TOOLS, we imho should only fail, if all of them are installed.

Reply:
hmm, so what do we gain from this regression test at all given that one missing tool is already sufficient?
main purpose and root cause for all of that was to verify that the development tools don't get included into release images by accident, with this softened criterion we could just as well do without that test imho.

Reply:
i disagree. @JanZachmann

Reply:
if we really want to prepare for this - to me at the moment theoretical - situation right now, instead of soften up the check to the verge of usefulness I would propose a variable defining allowed development tools, something like OMNECT_OS_DEVEL_TOOLS_ALLOWED, which is taken into account during negative checks.
with that approach we would keep control of what will get introduced into release images while retaining a useful test.
just my 0,02 €.

Reply:
I understand that the test isn't able to deal with tools that are already deps of other packages and those are even different between BPS's. Or is there a another weak point?
Imho I would prefer to live with the test anyway since:
- I also think the case is very unlikely
- I expect that we will have tools container images soon. thus we'll be able to remove all the tools handling and testing from meta-omnect
- In case I'm wrong regarding both upper points, we would need to refine/refactor the test

Reply:
rework complete, please review